hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/HiveSyncTool.java

```diff
@@ -251,10 +251,8 @@ private boolean syncSchema(String tableName, boolean tableExists, boolean useRea
         LOG.info("Schema difference found for " + tableName);
         hoodieHiveClient.updateTableDefinition(tableName, schema);
         // Sync the table properties if the schema has changed
-        if (cfg.tableProperties != null) {
-          hoodieHiveClient.updateTableProperties(tableName, tableProperties);
-          LOG.info("Sync table properties for " + tableName + ", table properties is: " + cfg.tableProperties);
-        }
+        hoodieHiveClient.updateTableProperties(tableName, tableProperties);
+        LOG.info("Sync table properties for " + tableName + ", table properties is: " + cfg.tableProperties);
```

Comment:
cfg.tableProperties ？，  i think it should be tableProperties

Reply:
how about change this code to：
        if (cfg.tableProperties != null || cfg.syncAsSparkDataSourceTable) {
          hoodieHiveClient.updateTableProperties(tableName, tableProperties);
          LOG.info("Sync table properties for " + tableName + ", table properties is: "
                  + (cfg.tableProperties == null ? "" : cfg.tableProperties));
        }

Reply:
if add new columns,and cfg.tableProperties is null,then do not executeupdateTableProperties,then spark sql will not get the new columns.
I'm not sure if delete columns and update columns are the same.
If not, I think it can be judged by `schemaDiff.getAddColumnTypes().isEmpty()`.

Reply:
> how about change this code to： if (cfg.tableProperties != null || cfg.syncAsSparkDataSourceTable) { hoodieHiveClient.updateTableProperties(tableName, tableProperties); LOG.info("Sync table properties for " + tableName + ", table properties is: " + (cfg.tableProperties == null ? "" : cfg.tableProperties)); }

Sorry to see this new news now. Let me think about it first

Reply:
no need to use chemaDiff.getAddColumnTypes().isEmpty().  your modify is ok, just 
pay attention to that: cfg.tableProperties maybe null and only if sync DataSourceTable we need these logical

Reply:
OK, I see. Thank you for your reminder. Your idea is better

Reply:
How about changing the log like this?
```java
LOG.info("Sync table properties for " + tableName + ", table properties is: " + tableProperties);
```

Reply:
I have submitted the newly modified code